disassemblers/ofrak_pyghidra/ofrak_pyghidra/standalone/pyghidra_analysis.py

```diff
@@ -0,0 +1,390 @@
+                        or (bb["virtual_address"] + bb["size"]) > cb["virtual_address"] + cb["size"]
+                    ):
+                        logging.warning(
+                            f"Basic Block 0x{bb['virtual_address']:x} does not fall within "
+                            f"complex block {hex(cb['virtual_address'])}-{hex(cb['virtual_address'] + cb['size'])}"
+                        )
+                        continue
+                    bb_key = f"bb_{bb['virtual_address']}"
+                    instructions = _unpack_basic_block(block, flat_api)
+                    bb["children"] = []
+                    for instruction in instructions:
+                        instr_key = f"instr_{instruction['virtual_address']}"
+                        bb["children"].append(instr_key)
+                        main_dictionary[instr_key] = instruction
+                    cb["children"].append(bb_key)
+                    main_dictionary[bb_key] = bb
+                for dw in data_words:
+                    if (
+                        dw["virtual_address"] < cb["virtual_address"]
+                        or (dw["virtual_address"] + dw["size"]) > cb["virtual_address"] + cb["size"]
+                    ):
+                        logging.warning(
+                            f"Data Word 0x{dw['virtual_address']:x} does not fall within "
+                            f"complex block {hex(cb['virtual_address'])}-{hex(cb['virtual_address'] + cb['size'])}"
+                        )
+                        continue
+                    dw_key = f"dw_{dw['virtual_address']}"
+                    cb["children"].append(dw_key)
+                    main_dictionary[dw_key] = dw
+                main_dictionary[cb_key] = cb
+    return main_dictionary
+
+
+def _unpack_program(flat_api):
+    ghidra_code_regions = []
+    for memory_block in flat_api.getMemoryBlocks():
+        is_execute = False
+        if memory_block.isExecute():
+            is_execute = True
+        vaddr = _parse_offset(memory_block.getStart())
+        size = memory_block.getSize()
+        ghidra_code_regions.append(
+            {"virtual_address": vaddr, "size": size, "executable": is_execute}
+        )
+    return _concat_contiguous_code_blocks(ghidra_code_regions)
+
+
+def _concat_contiguous_code_blocks(code_regions):
+    for i in range(len(code_regions) - 1):
+        if (
+            code_regions[i]["virtual_address"] + code_regions[i]["size"]
+            == code_regions[i + 1]["virtual_address"]
+            and code_regions[i]["executable"]
+            and code_regions[i + 1]["executable"]
+        ):
+            vaddr = code_regions[i]["virtual_address"]
+            size = code_regions[i]["size"] + code_regions[i + 1]["size"]
+            code_regions[i] = {"virtual_address": vaddr, "size": size, "executable": True}
+            del code_regions[i + 1]
+            return _concat_contiguous_code_blocks(code_regions)
+    return code_regions
```

Comment:
Seems like this will only work if the code regions are guaranteed to be sorted by virtual address, but I don't see us do that anywhere?

Reply:
This was a sort of hack due to a bug in ghidra where it split one code region into many of size 0x1000. I'm not totally sure if this is the right way to do this or if it should remain here, it could cause issues in other cases potentially.